draft-aegis-mimi-arch-latest.md

```diff
@@ -560,17 +564,38 @@ Output:
 
 The above request has the following effect:
 
-* An arbitrary key pool with attributes matching the provided `appId`,
+// TODO: Should we really have isFinalKey or is that potentially a security
+problem and we should say that the pool should just return an error in the event
+there are no more keys.
```

Comment:
Make sense to me to have isFinalKey. It allows users to trade better availability vs. better security which seems like a good idea. Apps can set a default but clients can modify it if they want.

Reply:
I saw a lot of chatter on the mailing list where people seemed to think it's not like other protocols and it's inherently bad.